eventing/samples/github-source/README.md

```diff
@@ -0,0 +1,160 @@
+
+```shell
+kubectl apply -f https://knative-releases.storage.googleapis.com/eventing-sources/latest/release.yaml
+```
+
+### Create GitHub Tokens
+
+1.  Create a [personal access
+    token](https://github.com/settings/tokens) for GitHub that the
+    GitHub source can use to register webhooks with the GitHub
+    API. Also decide on a secret token that your code will use to
+    authenticate the incoming webhooks from GitHub (*secretToken*).
+    
+    The token can be named anything you find convenient. The Source
+    requires `repo:public_repo` and `admin:repo_hook`, to let it fire
+    events from your public repositories and to create webhooks for
+    those repositories. Copy and save this token; GitHub will force
+    you to generate it again if misplaced.
+
+    Here's an example for a token named "GitHubSource Sample" with the
+    the recommended scopes:
+
+    ![GitHub UI](personal_access_token.png "GitHub personal access token screenshot")
+
+    Update `githubsecret.yaml` with those
+    values. If  your generated access token is `'asdfasfdsaf'` and you choose
+    your *secretToken* as `'personal_access_token_value'`, you'd modify
+    `githubsecret.yaml` like so:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: githubsecret
+type: Opaque
+stringData:
+  accessToken: asdfasfdsaf
+  secretToken: personal_access_token_value
+```
+
+Hint: you can makeup a random *accessToken* with:
+    
+```shell
+head -c 8 /dev/urandom | base64
+```
+
+Then, apply the githubsecret using `kubectl`:
+    
+```shell
+kubectl -n default apply -f eventing/samples/github-source/githubsecret.yaml
+```
+
+### Create Event Source for GitHub Events
+
+1. In order to receive GitHub events, you have to create a concrete
+   Event Source for a specific namespace. Be sure to replace the
+   `ownerAndRepository` value with a valid GitHub public repository
+   owned by your GitHub user. If you are using a different `Secret`
+   name, `Channel`, or `Service Account`, modify the yaml accordingly.
+
+```shell
+kubectl -n default apply -f eventing/samples/github-source/github-source.yaml
+```
+
+### Subscription
+
+To verify the `GitHubSource` is fully working, create a simple Knative
+Service that dumps incoming messages to its log and create a
+`Subscription` from the `Channel` to that Knative Service.
+
+1. If the deployed `GitHubEventSource` is pointing at a `Channel`
+   other than `githubchannel`, modify `subscription.yaml` by replacing
+   `githubchannel` with that `Channel`'s name.
+1. Deploy `subscription.yaml`.
+
+```shell
+ko apply -f eventing/samples/github-source/subscription.yaml
+```
+
+### Create Events
+
+Create a Pull Request in your GitHub repository.
+
+### Verify
+
+Verify the GitHub webhook was created by looking at the list of
+webhooks under the Settings tab in your GitHub repository. A hook
+should be listed that points to your Knative cluster with a green
+check mark to the left of the hook URL.
```

Comment:
Do you want a screenshot here? It seems like it would be helpful when describing the web UI.